From ce533ee265f24220649b2e07a8c90e1ea6deafb5 Mon Sep 17 00:00:00 2001
From: Wenyou Yang <wenyou.yang@atmel.com>
Date: Mon, 9 Feb 2015 09:21:12 +0800
Subject: [PATCH] board: move the duplicated function prototypes to board.h

Move the duplicated function prototypes from each board header file to the board.h.

Signed-off-by: Wenyou Yang <wenyou.yang@atmel.com>
Acked-by: Nicolas Ferre <nicolas.ferre@atmel.com>
---
 board/at91sam9260ek/at91sam9260ek.h       |  7 -------
 board/at91sam9261ek/at91sam9261ek.h       |  9 ---------
 board/at91sam9263ek/at91sam9263ek.h       |  9 ---------
 board/at91sam9g10ek/at91sam9g10ek.h       | 10 ----------
 board/at91sam9g20ek/at91sam9g20ek.h       |  9 ---------
 board/at91sam9m10g45ek/at91sam9m10g45ek.h | 10 ----------
 board/at91sam9n12ek/at91sam9n12ek.h       |  9 ---------
 board/at91sam9rlek/at91sam9rlek.h         |  8 --------
 board/at91sam9x5ek/at91sam9x5ek.h         |  9 ---------
 board/at91sam9xeek/at91sam9xeek.h         |  7 -------
 board/sama5d3_xplained/sama5d3_xplained.h |  9 ---------
 board/sama5d3xek/sama5d3xek.h             | 11 -----------
 board/sama5d4_xplained/sama5d4_xplained.h |  9 ---------
 board/sama5d4ek/sama5d4ek.h               |  9 ---------
 include/board.h                           | 13 +++++++++++++
 15 files changed, 13 insertions(+), 125 deletions(-)

diff --git a/board/at91sam9260ek/at91sam9260ek.h b/board/at91sam9260ek/at91sam9260ek.h
index 9f82a06..728bb6a 100644
--- a/board/at91sam9260ek/at91sam9260ek.h
+++ b/board/at91sam9260ek/at91sam9260ek.h
@@ -80,11 +80,4 @@
 #define CONFIG_SYS_RECOVERY_BUTTON_PIN  AT91C_PIN_PA(31)
 #define RECOVERY_BUTTON_NAME            "BP4"
 
-/* function */
-extern void hw_init(void);
-
-extern void nandflash_hw_init(void);
-
-extern void at91_spi0_hw_init(void);
-
 #endif /* __AT91SAM9260EK_H__ */
diff --git a/board/at91sam9261ek/at91sam9261ek.h b/board/at91sam9261ek/at91sam9261ek.h
index c9ca586..92e112b 100644
--- a/board/at91sam9261ek/at91sam9261ek.h
+++ b/board/at91sam9261ek/at91sam9261ek.h
@@ -99,13 +99,4 @@
 #define CONFIG_SYS_RECOVERY_BUTTON_PIN	AT91C_PIN_PA(26)
 #define RECOVERY_BUTTON_NAME	"BP4"
 
-/* function */
-extern void hw_init(void);
-
-extern void nandflash_hw_init(void);
-
-extern void at91_spi0_hw_init(void);
-
-extern void at91_mci0_hw_init(void);
-
 #endif /* #ifndef __AT91SAM9261EK_H__ */
diff --git a/board/at91sam9263ek/at91sam9263ek.h b/board/at91sam9263ek/at91sam9263ek.h
index bdeb65b..f29628c 100644
--- a/board/at91sam9263ek/at91sam9263ek.h
+++ b/board/at91sam9263ek/at91sam9263ek.h
@@ -112,13 +112,4 @@
 #define CONFIG_SYS_RECOVERY_BUTTON_PIN	AT91C_PIN_PC(4)
 #define RECOVERY_BUTTON_NAME	"BP2"
 
-/* export function */
-extern void hw_init(void);
-
-extern void nandflash_hw_init(void);
-
-extern void at91_spi0_hw_init(void);
-
-extern void at91_mci0_hw_init(void);
-
 #endif /* #ifndef __AT91SAM9263EK_H__ */
diff --git a/board/at91sam9g10ek/at91sam9g10ek.h b/board/at91sam9g10ek/at91sam9g10ek.h
index 12f339e..40315d3 100644
--- a/board/at91sam9g10ek/at91sam9g10ek.h
+++ b/board/at91sam9g10ek/at91sam9g10ek.h
@@ -110,14 +110,4 @@
 #define CONFIG_SYS_RECOVERY_BUTTON_PIN	AT91C_PIN_PA(26)
 #define RECOVERY_BUTTON_NAME	"BP4"
 
-/* function */
-/* function */
-extern void hw_init(void);
-
-extern void nandflash_hw_init(void);
-
-extern void at91_spi0_hw_init(void);
-
-extern void at91_mci0_hw_init(void);
-
 #endif /* #ifndef __AT91SAM9G10EK_H__*/
diff --git a/board/at91sam9g20ek/at91sam9g20ek.h b/board/at91sam9g20ek/at91sam9g20ek.h
index ae3b65a..4d832a7 100644
--- a/board/at91sam9g20ek/at91sam9g20ek.h
+++ b/board/at91sam9g20ek/at91sam9g20ek.h
@@ -84,13 +84,4 @@
 #define CONFIG_SYS_RECOVERY_BUTTON_PIN	AT91C_PIN_PA(31)
 #define RECOVERY_BUTTON_NAME	"BP4"
 
-/* function */
-extern void hw_init(void);
-
-extern void nandflash_hw_init(void);
-
-extern void at91_spi0_hw_init(void);
-
-extern void at91_mci0_hw_init(void);
-
 #endif	/* #ifndef __AT91SAM9G20EK_H__ */
diff --git a/board/at91sam9m10g45ek/at91sam9m10g45ek.h b/board/at91sam9m10g45ek/at91sam9m10g45ek.h
index d23e66b..7d87d19 100644
--- a/board/at91sam9m10g45ek/at91sam9m10g45ek.h
+++ b/board/at91sam9m10g45ek/at91sam9m10g45ek.h
@@ -108,14 +108,4 @@
 #define CONFIG_SYS_RECOVERY_BUTTON_PIN	AT91C_PIN_PB(7)
 #define RECOVERY_BUTTON_NAME	"BP4"
 
-/* function */
-extern void hw_init(void);
-
-extern void nandflash_hw_init(void);
-extern unsigned int nandflash_get_ready_pin(void);
-
-extern void at91_spi0_hw_init(void);
-
-extern void at91_mci0_hw_init(void);
-
 #endif /* __AT91SAM9M10G45EK_H__ */
diff --git a/board/at91sam9n12ek/at91sam9n12ek.h b/board/at91sam9n12ek/at91sam9n12ek.h
index c2871a7..e39a820 100644
--- a/board/at91sam9n12ek/at91sam9n12ek.h
+++ b/board/at91sam9n12ek/at91sam9n12ek.h
@@ -104,13 +104,4 @@
 #define CONFIG_SYS_RECOVERY_BUTTON_PIN	AT91C_PIN_PB(3)
 #define RECOVERY_BUTTON_NAME	"PB4"
 
-/* function */
-extern void hw_init(void);
-
-extern void nandflash_hw_init(void);
-
-extern void at91_spi0_hw_init(void);
-
-extern void at91_mci0_hw_init(void);
-
 #endif /* #ifndef __AT91SAM9N12EK_H__ */
diff --git a/board/at91sam9rlek/at91sam9rlek.h b/board/at91sam9rlek/at91sam9rlek.h
index 66c274a..eb8fe78 100644
--- a/board/at91sam9rlek/at91sam9rlek.h
+++ b/board/at91sam9rlek/at91sam9rlek.h
@@ -94,12 +94,4 @@
 #define CONFIG_SYS_RECOVERY_BUTTON_PIN	AT91C_PIN_PB(0)
 #define RECOVERY_BUTTON_NAME	"BP2"
 
-/* extern function */
-extern void hw_init(void);
-
-extern void nandflash_hw_init(void);
-
-extern void at91_spi0_hw_init(void);
-
-extern void at91_mci0_hw_init(void);
 #endif /* #ifndef __AT91SAM9RLEK_H__ */
diff --git a/board/at91sam9x5ek/at91sam9x5ek.h b/board/at91sam9x5ek/at91sam9x5ek.h
index 5eacf82..37f39f0 100644
--- a/board/at91sam9x5ek/at91sam9x5ek.h
+++ b/board/at91sam9x5ek/at91sam9x5ek.h
@@ -116,13 +116,4 @@
  */
 #define CONFIG_SYS_ONE_WIRE_PIN		AT91C_PIN_PB(18)
 
-/* function */
-extern void hw_init(void);
-
-extern void nandflash_hw_init(void);
-
-extern void at91_spi0_hw_init(void);
-
-extern void at91_mci0_hw_init(void);
-
 #endif /*#ifndef __AT91SAM9X5EK_H__ */
diff --git a/board/at91sam9xeek/at91sam9xeek.h b/board/at91sam9xeek/at91sam9xeek.h
index c605b0a..1a7d457 100644
--- a/board/at91sam9xeek/at91sam9xeek.h
+++ b/board/at91sam9xeek/at91sam9xeek.h
@@ -74,11 +74,4 @@
 
 #define CONFIG_SYS_NAND_ENABLE_PIN	AT91C_PIN_PC(14)
 
-/* export function */
-extern void hw_init(void);
-
-extern void nandflash_hw_init(void);
-
-extern void at91_spi0_hw_init(void);
-
 #endif	/* #ifndef __AT91SAM9XEEK_H__ */
diff --git a/board/sama5d3_xplained/sama5d3_xplained.h b/board/sama5d3_xplained/sama5d3_xplained.h
index 59b5fec..aeb4fe1 100644
--- a/board/sama5d3_xplained/sama5d3_xplained.h
+++ b/board/sama5d3_xplained/sama5d3_xplained.h
@@ -213,13 +213,4 @@
  */
 #define CONFIG_SYS_BASE_MCI	AT91C_BASE_HSMCI0
 
-/* function */
-extern void hw_init(void);
-
-extern void nandflash_hw_init(void);
-
-extern void at91_spi0_hw_init(void);
-
-extern void at91_mci0_hw_init(void);
-
 #endif /* __SAMA5D3_XPLAINED_H__ */
diff --git a/board/sama5d3xek/sama5d3xek.h b/board/sama5d3xek/sama5d3xek.h
index ce8d14f..e7547a6 100644
--- a/board/sama5d3xek/sama5d3xek.h
+++ b/board/sama5d3xek/sama5d3xek.h
@@ -245,15 +245,4 @@
  */
 #define CONFIG_SYS_ONE_WIRE_PIN		AT91C_PIN_PE(25)
 
-/* function */
-extern void hw_init(void);
-
-extern void nandflash_hw_init(void);
-
-extern void at91_spi0_hw_init(void);
-
-extern void at91_mci0_hw_init(void);
-
-extern void norflash_hw_init(void);
-
 #endif /* #ifndef __SAMA5EK_H__ */
diff --git a/board/sama5d4_xplained/sama5d4_xplained.h b/board/sama5d4_xplained/sama5d4_xplained.h
index cacf051..1840e9d 100644
--- a/board/sama5d4_xplained/sama5d4_xplained.h
+++ b/board/sama5d4_xplained/sama5d4_xplained.h
@@ -181,13 +181,4 @@
  */
 #define CONFIG_SYS_HDMI_RESET_PIN	AT91C_PIN_PB(15)
 
-/* function */
-extern void hw_init(void);
-
-extern void nandflash_hw_init(void);
-
-extern void at91_spi0_hw_init(void);
-
-extern void at91_mci0_hw_init(void);
-
 #endif
diff --git a/board/sama5d4ek/sama5d4ek.h b/board/sama5d4ek/sama5d4ek.h
index df830d7..8d7f570 100644
--- a/board/sama5d4ek/sama5d4ek.h
+++ b/board/sama5d4ek/sama5d4ek.h
@@ -192,13 +192,4 @@
  */
 #define CONFIG_SYS_HDMI_RESET_PIN	AT91C_PIN_PB(15)
 
-/* function */
-extern void hw_init(void);
-
-extern void nandflash_hw_init(void);
-
-extern void at91_spi0_hw_init(void);
-
-extern void at91_mci0_hw_init(void);
-
 #endif
diff --git a/include/board.h b/include/board.h
index ef966da..7881c54 100644
--- a/include/board.h
+++ b/include/board.h
@@ -84,4 +84,17 @@
 #include "sama5d4_xplained.h"
 #endif
 
+/*
+ * Functions Prototype
+ */
+extern void hw_init(void);
+
+extern void nandflash_hw_init(void);
+
+extern void at91_spi0_hw_init(void);
+
+extern void at91_mci0_hw_init(void);
+
+extern void norflash_hw_init(void);
+
 #endif /* #ifndef __BOARD_H__ */
-- 
1.9.1

